cni/network/multitenancy.go

```diff
@@ -203,7 +208,77 @@ func (m *Multitenancy) GetContainerNetworkConfiguration(
 	return ncResponse, hostSubnetPrefix, err
 }
 
-func (m *Multitenancy) getContainerNetworkConfigurationInternal(
+// get all network container configurations by given orchestratorContext
+func (m *Multitenancy) GetNetworkContainersWithOrchestratorContext(
+	ctx context.Context, nwCfg *cni.NetworkConfig, podName, podNamespace string,
+) (*[]cns.GetNetworkContainerResponse, []net.IPNet, error) {
+	var podNameWithoutSuffix string
+
+	if !nwCfg.EnableExactMatchForPodName {
+		podNameWithoutSuffix = network.GetPodNameWithoutSuffix(podName)
+	} else {
+		podNameWithoutSuffix = podName
+	}
+
+	log.Printf("Podname without suffix %v", podNameWithoutSuffix)
+
+	ncResponses, hostSubnetPrefix, err := m.getNetworkContainersWithOrchestratorContextInternal(ctx, podNamespace, podNameWithoutSuffix)
+
+	responses := *ncResponses
+	if ncResponses != nil {
+		for i := 0; i < len(responses); i++ {
+			if nwCfg.EnableSnatOnHost {
+				if responses[i].LocalIPConfiguration.IPSubnet.IPAddress == "" {
+					log.Printf("Snat IP is not populated. Got empty string")
+					return nil, []net.IPNet{}, errSnatIP
+				}
+			}
+		}
+	}
+	return ncResponses, hostSubnetPrefix, err
```

Comment:
As mentioned above I think we should just return slice instead of pointer to slice

Reply:
+1 to this. Slices are basically fat pointers anyway, so adding indirection to them is usually no savings (it just makes things harder to read).